<!doctype html><html lang=ja dir=ltr class=dark><head><meta charset=utf-8><meta name=viewport content="width=device-width"><meta name=generator content="Hugo 0.124.1"><meta property="og:title" content='pip実行時にsys.stderr.write(f"ERROR: {exc} ") とエラーが出たときの対処法'><meta property="og:description" content='pip実行時にsys.stderr.write(f"ERROR: {exc} ") とエラーが出たときの対処法'><meta property="og:type" content="article"><meta property="og:url" content="https://zEttOn86.github.io/blog/20210422-remove-pip-error/"><meta property="og:image" content="https://zEttOn86.github.io/images/blog/210422/coffee.png"><meta property="article:section" content="blog"><meta property="article:published_time" content="2021-04-22T22:22:22+09:00"><meta property="article:modified_time" content="2025-05-01T23:25:26+09:00"><title>pip実行時にsys.stderr.write(f"ERROR: {exc} ") とエラーが出たときの対処法 | まぁそういう時もある。
</title><meta name=author content="zEttOn86"><meta name=description content='
    pip実行時にsys.stderr.write(f"ERROR: {exc} ") とエラーが出たときの対処法
  '><meta name=keywords content="
    pip,python3.5
  "><link rel=icon href=https://zEttOn86.github.io/images/favicon.ico><link rel=preload href="https://zEttOn86.github.io/css/style.min.4ceb331dc3dc3f3825899042590ee327a24c7acdabfd7f686c40e52e9ef8220a.css" as=style onload='this.onload=null,this.rel="stylesheet"' integrity="sha256-TOszHcPcPzgliZBCWQ7jJ6JMes2r/X9obEDlLp74Igo=" crossorigin=anonymous><noscript><link rel=stylesheet href="https://zEttOn86.github.io/css/style.min.4ceb331dc3dc3f3825899042590ee327a24c7acdabfd7f686c40e52e9ef8220a.css"></noscript><link rel=stylesheet href="https://zEttOn86.github.io/css/main.min.14fa8d80e4d6a72f4bbaba11d6caf4aa6069b88549b3b08166772c1bab5c871f.css" integrity="sha256-FPqNgOTWpy9LuroR1sr0qmBpuIVJs7CBZncsG6tchx8=" crossorigin=anonymous><link rel=stylesheet href="https://zEttOn86.github.io/css/custom.min.0c83eb17d2d537e6a49cea7ccde082a914f974215cd38170a93d658c749014f8.css" integrity="sha256-DIPrF9LVN+aknOp8zeCCqRT5dCFc04FwqT1ljHSQFPg=" crossorigin=anonymous></head><body class="flex min-h-screen w-full flex-col"><header class="z-10 flex flex-none justify-center"><div class="relative flex h-20 w-full items-center justify-between px-5 md:mt-3 md:w-3/5 md:px-0 lg:max-w-5xl"><div class=flex-none><a href=https://zEttOn86.github.io/><h1 class=text-2xl>まぁそういう時もある。</h1></a></div><nav class=relative><ul class="hidden space-x-4 md:flex"><li><a class=hover:underline href=/about>About</a></li><li><a class=hover:underline href=/static/disclaimers>Disclaimers</a></li><li><a class=hover:underline href=/static/privacy-policy>Privacy Policy</a></li></ul><button id=menuBtn type=button title="Menu button" class="relative z-10 flex rounded-md border-none bg-slate-800 md:hidden">
<i id=menuIcon class="fa-solid fa-bars fa-2x"></i>
<i id=menuIconX class="fa-solid fa-xmark fa-2x hidden"></i></button><ul id=hamburgerMenu class="fixed right-0 top-0 z-0 flex w-screen -translate-y-full transform flex-col justify-end space-y-3 bg-[#252525] pb-10 transition-transform duration-150 md:hidden"><li class="flex w-full justify-center py-4 hover:bg-[#202020]"><a class=hover:underline href=/about>About</a></li><li class="flex w-full justify-center py-4 hover:bg-[#202020]"><a class=hover:underline href=/static/disclaimers>Disclaimers</a></li><li class="flex w-full justify-center py-4 hover:bg-[#202020]"><a class=hover:underline href=/static/privacy-policy>Privacy Policy</a></li></ul></nav></div></header><main class="flex flex-auto justify-center"><div class="max-w-4xl md:w-3/5 lg:max-w-5xl"><div id=articleMain class="mx-2 mt-6 flex flex-col overflow-hidden rounded-lg bg-gray-700 shadow-md md:mx-0"><div><a href=/blog/20210422-remove-pip-error/></a></div><div class="flex flex-col gap-y-3 p-6"><h1 class="text-4xl font-semibold text-slate-100"><a href=/blog/20210422-remove-pip-error/>pip実行時にsys.stderr.write(f"ERROR: {exc} &ldquo;) とエラーが出たときの対処法</a></h1><h2 class="my-4 text-lg text-slate-300">pip実行時にsys.stderr.write(f"ERROR: {exc} &ldquo;) とエラーが出たときの対処法</h2><ul class="flex flex-row flex-wrap text-slate-300"><li><a href=/categories/%E6%8A%80%E8%A1%93%E7%B3%BB/ class="mr-2 rounded border border-emerald-800 bg-emerald-800 px-2 py-1 text-sm text-slate-50">技術系</a></li><li><a href=/tags/pip/ class="mr-2 flex flex-row py-1 text-sm hover:font-bold"><i class="fa fa-tags"></i><span class="hover-underline-blue ml-1 text-link-blue hover:underline">pip</span></a></li><li><a href=/tags/python3.5/ class="mr-2 flex flex-row py-1 text-sm hover:font-bold"><i class="fa fa-tags"></i><span class="hover-underline-blue ml-1 text-link-blue hover:underline">python3.5</span></a></li></ul><span class=lastmod><li class=list-none><i class="fas fa-pencil-alt"></i>
<time datetime=2021-04-22T22:22:22JST>2021/04/22</time></li><li class=list-none><i class="fas fa-sync-alt"></i>
<time datetime=2025-05-01T23:25:26JST>2025/05/01</time></li></span><section class="article-toc w-full max-w-4xl rounded-lg bg-gray-600 p-4 text-slate-100 shadow-md lg:max-w-5xl"><h2 class=text-2xl>目次</h2><aside><nav id=TableOfContents><ul><li><ul><li><a href=#環境>環境</a></li><li><a href=#対処方法>対処方法</a></li><li><a href=#そもそもの対処法>そもそもの対処法</a></li><li><a href=#参考文献>参考文献</a></li></ul></li></ul></nav></aside></section><article class="prose-quoteless article-content prose prose-invert mt-6 w-full max-w-4xl break-words text-slate-300 lg:max-w-5xl"><p>一年に一回は遭遇している気がするのでメモしておきます。</p><h3 id=環境>環境</h3><ul><li>Ubuntu16.04</li><li>python3.5.2</li></ul><h3 id=対処方法>対処方法</h3><p><code>pip install --upgrade pip</code>を実行した後に<code>pip</code>を実行すると以下のようなエラーがでることがあります。</p><div class=highlight><pre tabindex=0 class=chroma><code class=language-bash data-lang=bash><span class=line><span class=cl>Traceback <span class=o>(</span>most recent call last<span class=o>)</span>:
</span></span><span class=line><span class=cl>  File <span class=s2>&#34;get-pip.py&#34;</span>, line 24226, in &lt;module&gt;
</span></span><span class=line><span class=cl>    main<span class=o>()</span>
</span></span><span class=line><span class=cl>  File <span class=s2>&#34;get-pip.py&#34;</span>, line 199, in main
</span></span><span class=line><span class=cl>    bootstrap<span class=o>(</span><span class=nv>tmpdir</span><span class=o>=</span>tmpdir<span class=o>)</span>
</span></span><span class=line><span class=cl>  File <span class=s2>&#34;get-pip.py&#34;</span>, line 82, in bootstrap
</span></span><span class=line><span class=cl>    from pip._internal.cli.main import main as pip_entry_point
</span></span><span class=line><span class=cl>  File <span class=s2>&#34;/tmp/tmpUgc5ng/pip.zip/pip/_internal/cli/main.py&#34;</span>, line <span class=m>60</span>
</span></span><span class=line><span class=cl>    sys.stderr.write<span class=o>(</span>f<span class=s2>&#34;ERROR: {exc}&#34;</span><span class=o>)</span>
</span></span><span class=line><span class=cl>                                   ^
</span></span><span class=line><span class=cl>
</span></span><span class=line><span class=cl>SyntaxError: invalid syntax
</span></span></code></pre></div><p>これは、<code>f""</code>という記述がpython3.5ではサポートされていないことが原因です。python3.6からこの記述はサポートされています。<br>Ubuntuには、デフォルトでpython2.X系が入っているので、python3.X系、python2.X系の両方を直します。<br>まず、python2.X系の対処法です。</p><div class=highlight><pre tabindex=0 class=chroma><code class=language-bash data-lang=bash><span class=line><span class=cl>$ wget https://bootstrap.pypa.io/pip/2.7/get-pip.py
</span></span><span class=line><span class=cl>$ python get-pip.py
</span></span></code></pre></div><p>つぎに、python3.X系の対処法です。<br>上記と同様に実行すると以下のようなエラーに遭遇したため、<code>/usr/bin/pip3</code>を削除してからインストールしました。</p><div class=highlight><pre tabindex=0 class=chroma><code class=language-bash data-lang=bash><span class=line><span class=cl>$ python get-pip.py
</span></span><span class=line><span class=cl>Traceback <span class=o>(</span>most recent call last<span class=o>)</span>:
</span></span><span class=line><span class=cl>  File <span class=s2>&#34;get-pip.py&#34;</span>, line 24, in &lt;module&gt;
</span></span><span class=line><span class=cl>    import pkgutil
</span></span><span class=line><span class=cl>ImportError: No module named pkgutil
</span></span></code></pre></div><p>以下が手順です。</p><div class=highlight><pre tabindex=0 class=chroma><code class=language-bash data-lang=bash><span class=line><span class=cl>$ sudo apt-get --purge autoremove python3-pip
</span></span><span class=line><span class=cl>$ wget https://bootstrap.pypa.io/pip/3.5/get-pip.py
</span></span><span class=line><span class=cl>$ python3 get-pip.py 
</span></span></code></pre></div><h3 id=そもそもの対処法>そもそもの対処法</h3><p><code>pip install --upgrade pip</code>をする際に、サポートされているバージョンを以下のように指定していれば、このようなエラーは発生しないと思います。</p><div class=highlight><pre tabindex=0 class=chroma><code class=language-python data-lang=python><span class=line><span class=cl><span class=err>$</span> <span class=n>pip3</span> <span class=n>install</span> <span class=o>--</span><span class=n>upgrade</span> <span class=n>pip</span><span class=o>==</span><span class=mf>20.3</span>
</span></span></code></pre></div><h3 id=参考文献>参考文献</h3><ul><li><a href=https://shunyaueta.com/posts/2021-02-08/>pip 実行時に sys.stderr.write(f"ERROR: {exc} &ldquo;) とエラーが出てpipを実行できないときの対処方法</a></li><li><a href=https://stackoverflow.com/questions/65869296/installing-pip-is-not-working-in-python-3-6>Installing pip is not working in python &lt; 3.6</a></li><li><a href=https://stackoverflow.com/questions/43847317/cant-remove-python-pip>can&rsquo;t remove python pip</a></li><li><a href=https://stackoverflow.com/questions/50535136/python-pip-install-missing-module-pkgutil>Python pip install missing module pkgutil [closed]</a></li></ul></article><div class="justify-between text-slate-100 md:flex md:flex-row"><a class="block hover:font-bold hover:underline" href=https://zEttOn86.github.io/blog/20220101-how-to-install-hugo-via-homebrew/>&#171; MEMO: How to Install Hugo via Homebrew on WSL</a>
<a class="block hover:font-bold hover:underline md:text-right" href=https://zEttOn86.github.io/blog/20210407-ssh-setting/>SSH設定のメモ &#187;</a></div><div id=comments class="flex flex-col px-5 md:mt-3 md:px-0"><div id=disqus_thread></div><script>window.disqus_config=function(){},function(){if(["localhost","127.0.0.1"].indexOf(window.location.hostname)!=-1){document.getElementById("disqus_thread").innerHTML="Disqus comments not available by default when the website is previewed locally.";return}var t=document,e=t.createElement("script");e.async=!0,e.src="//learning-notes-1.disqus.com/embed.js",e.setAttribute("data-timestamp",+new Date),(t.head||t.body).appendChild(e)}()</script><noscript>Please enable JavaScript to view the <a href=https://disqus.com/?ref_noscript>comments powered by Disqus.</a></noscript><a href=https://disqus.com class=dsq-brlink>comments powered by <span class=logo-disqus>Disqus</span></a></div></div></div></div></main><footer class="flex flex-auto justify-center"><section class="py-6 text-lg dark:text-slate-300 md:py-3">&copy; Copyright
2019, zEttOn86 - All rights reserved</section></footer><script src=https://zEttOn86.github.io/js/main.2c93a4001bbafe7a459848d96fbcecbb3bd281b24ff896e018d81ae277c9d4a4.js integrity="sha256-LJOkABu6/npFmEjZb7zsuzvSgbJP+JbgGNga4nfJ1KQ=" crossorigin=anonymous></script><script>MathJax={tex:{displayMath:[["\\[","\\]"],["$$","$$"]],inlineMath:[["$","$"],["\\(","\\)"]]}}</script><script defer type=text/javascript id=MathJax-script src=https://zEttOn86.github.io/mathjax/es5/tex-mml-chtml.b6268d980ce9cb729c82dba22f04fd592952b2a1aab43079ca5330c68a86e72b0d232ce4070db893a5054ee5c68325c92c9f1a33f868d61ebb35129e74fc7ef9.js integrity="sha512-tiaNmAzpy3KcgtuiLwT9WSlSsqGqtDB5ylMwxoqG5ysNIyzkBw24k6UFTuXGgyXJLJ8aM/ho1h67NRKedPx++Q=="></script><script async src="https://www.googletagmanager.com/gtag/js?id=G-9NRRDF48L8"></script><script>var doNotTrack=!1;if(!doNotTrack){window.dataLayer=window.dataLayer||[];function gtag(){dataLayer.push(arguments)}gtag("js",new Date),gtag("config","G-9NRRDF48L8",{anonymize_ip:!1})}</script></body></html>